lib/npo.src.js

```diff
@@ -77,6 +124,15 @@
 			cycle = timer(scheduling_queue.drain);
 		}
 	}
+	
+	function hasCatch(chain) {
+		for (var i = 0; i < chain.length; i++) {
+			if (chain[i].failure) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
[This code](https://github.com/zloirock/core-js/blob/v2.4.1/modules/es6.promise.js), you mean? That's definitely not NPO. But good point about tests. :)